Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/qrcode/VaccinationQRCodeExtractor.kt

```diff
@@ -56,6 +57,9 @@ class VaccinationQRCodeExtractor @Inject constructor(
             header = headerParser.parse(cbor),
             certificate = bodyParser.parse(cbor)
         ).also {
+            CertificateQrCodeCensor.dataToCensor =
```

Comment:
Can we move the logic into eg. the repo and out of the extractor? Extract will be called each time the content is retrieved and does not need to know about censoring.

Reply:
We need to set it as early as possible. Otherwise we rely on the repo actually storing it, which may or may not happen, and if it happens, but takes too long, we could still have logged it.

The issue is "data that we have to censor, but don't yet know we have to censor".

Reply:
The logger currently solves this by delaying any logged lines for a short duration, but we can't increase this delay endlessly as it consumes memory to buffer the log lines.

Reply:
Maybe in the view model then? Before and/or after the extraction?